modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/display/context/SortDisplayBuilder.java

```diff
@@ -70,6 +96,9 @@ protected SortTermDisplayContext buildTermDisplayContext(
 			new SortTermDisplayContext();
 
 		sortTermDisplayContext.setLabel(label);
+		sortTermDisplayContext.setLanguageLabel(
+			LanguageUtil.get(
+				PortalUtil.getHttpServletRequest(_renderRequest), label));
```

Comment:
We're trying to move away from these `*Util` classes. Can we try using the `Language` and `Portal` classes directly instead?

```java
language.get(portal.getHttpServletRequest(_renderRequest), label));
```

```java
@Reference
protected Language language;

@Reference
protected Portal portal;
```

Reply:
Doing this seems to cause a null pointer exception for me :\

Reply:
Ah, I think we'll have to pass these in then. In `SortPortlet.java` and `SortConfigurationAction.java` we can add:
```java
@Reference
protected Language language;

@Reference
protected Portal portal;
```

Then the `SortDisplayBuilder` will be:
```java
SortDisplayBuilder(language, portal, renderRequest, sortPortletPreferences)
```

Reply:
Ooo thank you! That worked